File: src/main/java/client/renderables/BlockRender.java
package client.renderables;

import client.Game;
import common.block.Block;

import java.awt.*;
import java.awt.image.BufferedImage;

public class BlockRender implements Entity, Drawable {
    private BufferedImage texture;
    private Block block;
    private Block lastBlock;
    public int x;
    public int width = 64;
    public int y;
    public int height = 64;

    public BlockRender(Block block, int x, int y){
        this.texture = Game.textures.getTexture4(block.getName());
        this.block = block;
        this.lastBlock = block;
        this.x = x;
        this.y = y;
        this.height = 64;
        this.width = 64;
    }

    @Override
    public void draw(Graphics g) {
        if (block != lastBlock){
            this.texture = Game.textures.getTexture4(block.getName());
        }
        g.drawImage(texture, x, y, Game.instance);
        lastBlock = block;
    }

    @Override
    public void tick(){
    }

    @Override
    public void second() {}

    public Block getBlock() {
        return block;
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public void setCoords(int i, int i1) {
        this.x = i;
        this.y = i1;
    }

    public void setBlock(Block block) {
        this.block = block;
    }
}

File: src/main/java/client/renderables/BlockRender.kt
package client.renderables

import common.block.Block
import java.awt.Graphics
import java.awt.image.BufferedImage

/* TODO: You don't need a render object per block at every position. Use one renderer and draw the same block at varying
 positions. */
class BlockRender(var block: Block?, var x: Int, var y: Int) : RenderedEntity(), Drawable {

    private var texture: BufferedImage? = null
    private var lastBlock: Block? = null
    var width = 64
    var height = 64

    init {
        //this.texture = Game.textures.getTexture4(block?.name)
        this.lastBlock = block
        this.height = 64
        this.width = 64
    }

    override fun draw(g: Graphics) {
        if (block !== lastBlock) {
            //this.texture = Game.textures.getTexture4(block?.name)
        }
        //g.drawImage(texture, x, y, Game.instance)
        lastBlock = block
    }

    override fun tick() {}

    override fun second() {}

    fun setCoords(i: Int, i1: Int) {
        this.x = i
        this.y = i1
    }
}
